src/realm/array_object_id.cpp

```diff
@@ -99,13 +99,24 @@ void ArrayObjectId::move(ArrayObjectId& dst_arr, size_t ndx)
 
     const auto n_to_move = old_src_size - ndx;
 
+    dst_arr.copy_on_write();
```

Comment:
I am not really sure if this is needed. Is it just to be sure? I would expect that the destination array would always be writeable. Can a test case be made where this is not true?
I you think it is needed, it should be the version where you specify the needed bytes. It is calculated below. This will ensure that we only allocate once.

Reply:
After looking into this deeper, I found that the call to `dst_arr.alloc(...)` below will call copy_on_write if necessary so this call is indeed not necessary. Thanks for pointing this out!